geotrek/api/v2/filters.py

```diff
@@ -717,12 +717,47 @@ def get_schema_fields(self, view):
         )
 
 
+class OutdoorRatingsFilter(BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        ratings = request.GET.get('ratings')
+        if ratings:
+            queryset = queryset.filter(ratings__in=ratings.split(','))
+        return queryset
+
+    def get_schema_fields(self, view):
+        return (
+            Field(
+                name='ratings', required=False, location='query', schema=coreschema.Integer(
+                    title=_("Ratings"),
+                    description=_('Filter by one or more ratings id, comma-separated.')
+                )
+            ),
+        )
+
+
 class GeotrekSiteFilter(BaseFilterBackend):
     def filter_queryset(self, request, queryset, view):
         root_sites_only = request.GET.get('root_sites_only')
         if root_sites_only:
             # Being a root node <=> having no parent
             queryset = queryset.filter(parent=None)
+        practices_in_hierarchy = request.GET.get('practices_in_hierarchy')
+        if practices_in_hierarchy:
+            wanted_practices = set(map(int, practices_in_hierarchy.split(',')))
+            queryset = queryset.filter('super_practices__id_in=wabted')
+            for site in queryset:
+                # Exclude if practices in hierarchy don't match any wanted practices
+                found_practices = site.super_practices_id
+                if found_practices.isdisjoint(wanted_practices):
+                    queryset = queryset.exclude(id=site.pk)
+        ratings_in_hierarchy = request.GET.get('ratings_in_hierarchy')
+        if ratings_in_hierarchy:
+            wanted_ratings = set(map(int, ratings_in_hierarchy.split(',')))
+            for site in queryset:
+                # Exclude if ratings in hierarchy don't match any wanted ratings
+                found_ratings = site.super_ratings_id
+                if found_ratings.isdisjoint(wanted_ratings):
+                    queryset = queryset.exclude(id=site.pk)
```

Comment:
Je sais que ce n'est pas idéal d'itérer sur le queryset, mais vu la complexité pour récupérer les pratiques et les cotations des sites enfants, je n'ai pas d'autre idées

Reply:
Ok, pour le moment on valide comme ça. Si ca pose des problèmes de perf on prendra le temps de réfléchir à une meilleure manière de faire alors.
Met juste un commentaire en TODO dans le code qu'on puisse savoir que c'est optimisable.